### ISSUE TITLE
Lesson "Simulating Memory" (Listings 51, 52, and 53)

### ISSUE THREAD

**Issue by hintron** [hintron/computer-enhance#8]
[Watch the next lesson](https://www.computerenhance.com/p/simulating-memory) about simulating memory and implement [listings](https://github.com/cmuratori/computer_enhance/tree/main/perfaware/part1) 51, 52, and 53.

This shouldn't be too hard - all I need to do is add a big memory array and access it during execution.

**Comment by hintron** [comment]
I refactored and reworked a lot of things in preparation for implementing listing 51 in commits 9a54b46...97b88de as 8-v1.

**Comment by hintron** [comment]
I implemented listing 51 in commits 97b88de633319c153ccc653c7339445f86d554c6...362d753 (mostly in commit 362d75353587ea00a50d5d87fab7aa8a0c2717e3).

**Comment by hintron** [comment]
I implemented listings 52 and 53 with commits 362d753...307a367. Marking this as implemented!

I had to modify listings 51, 52, and 53 slightly in order to match the output of my decoder. These listings are not consistent with the decoding listings. Things I had to change:
* I removed some unnecessary `word` prefixes for reg to mem accesses. I verified that it produces the same encoding either way.
* I spaced out `[bp+si]` to `[bp + si]`
* I modified some of the earlier simulation listings to not print the flags register when it's empty, since that is what these newer listings do.

These changes are all superficial and don't change the actual results of the listings.
